build_tools/third_party/cuda/CMakeLists.txt

```diff
@@ -0,0 +1,78 @@
+# Copyright 2022 The IREE Authors
```

Comment:
I'm not sure I would put this in `build_tools/third_party`. In this case it's just a script we have, so it's not like it's third_party code. We've got things a bit confused on that front though (e.g. the script for installing swiftshader is in third_party), but in my mind a "third_party" directory is for code we don't thing we own the license for or is super tightly coupled with the same (e.g. we might make some of our own modifications to the cmake file someone else has). IDK I don't think we've been super consistent. @ScottTodd thoughts?

Reply:
Again, echoing comment above, this is in service of #8971 which advocates for this design. I'm also following the patterns used by all other `build_tools/third_party` CMake scripts. 

Personally, if there is a grand philosophy for how we're organizing 3rd party dependencies I'm all ears? There is no clearcut information on how our build scripts should be organized so I'm following whats previously been done. 

There does seem to be some intended separation between the `third_party` CMake scripts and the rest of the code. They're only included `add_subdirectory()`. None of them expose functions to be used elsewhere in the code nor they explicitly return results, though they do define targets that can be referenced elsewhere.

Reply:
Seems reasonable to keep this where it is, following the SwiftShader example.

* `build_tools/third_party/` is for tools we write to help with third party dependencies - including both target/workspace definitions for Bazel/CMake and scripts that help with setup
* `third_party/` is for third party source code